lib/dams/controller_helper.rb

```diff
@@ -44,6 +44,71 @@ def osf_contributors(document)
 		  osf_data = (osf_data.blank?) ? osf_data << {"name": "UC San Diego Library"} : osf_data
 		end
 
+		def osf_related_agents(document)
+			field_name = "relationship_json_tesim"
+			dams_data = document["#{field_name}"]
+			osf_data =[]
+
+			if dams_data != nil
+		    dams_data.each do |datum|
+		    	
+		      relationships = JSON.parse(datum)
+		    	relationships.each do |key, value|
```

Comment:
You could potentially use `map` here instead of generate the array for you. But just a thought, this definitely works as-is.